src/server/raft_client.rs

```diff
@@ -848,10 +871,23 @@ async fn start<S, R, E>(
                         .with_label_values(&["unreachable", &back_end.store_id.to_string()])
                         .inc_by(1);
                 }
-                back_end
-                    .builder
-                    .router
-                    .broadcast_unreachable(back_end.store_id);
+
+                // broadcast is time consuming operation which would blocks raftstore, so report
+                // unreachable only once until being connected again.
+                if retry_times == 1 || ever_connected {
```

Comment:
This is a bit complicated. You can check `Channel::wait_for_connected`.

Reply:
It's not complicated. Only add a check `grpc_error_is_connect_err`, same as existing `grpc_error_is_unimplemented`. 
Moreover, using `wait_for_connected` I can't distinguish whether it's connected before but disconnected now or it's never connected

Reply:
```
let channel = backend.connect();
let connected = channel.wait_for_connected(deadline)?;
if !connected {
    // handle never connected
} else {
    let client = Client::new(channel);
    client.rpc_call();
}
```

Reply:
done